File: homework/ba-nguyen/lession-1/range-sum-query-immutable.js
/**
 * @param {number[]} numbs
 */
const NumArray = function (numbs) {
  this.prefixNumbs = [numbs[0]];

  for (let i = 1; i < numbs.length; i++) {
    this.prefixNumbs[i] = this.prefixNumbs[i - 1] + numbs[i];
  }
};

/**
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
NumArray.prototype.sumRange = function (left, right) {
  console.log(this.prefixNumbs[right], this.prefixNumbs[left]);
  return this.prefixNumbs[right] - (this.prefixNumbs[left - 1] || 0);
};

File: homework/quang_tu/lesson_1/range-sum-query-immutable.cpp
class NumArray {
public:
    vector<int> arr = {0};

    NumArray(vector<int>& nums) {
        for (int i = 0; i < nums.size(); ++i)
            arr.push_back(arr[i] + nums[i]);
    }

    int sumRange(int left, int right) {
        return arr[right + 1] - arr[left];
    }
};

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray* obj = new NumArray(nums);
 * int param_1 = obj->sumRange(left,right);
 */
